.config/eslint.mjs

```diff
@@ -0,0 +1,581 @@
+    'no-mixed-spaces-and-tabs': ON,
+    'no-multi-spaces': [ON, { ignoreEOLComments: true }],
+    'no-multiple-empty-lines': [ON, { max: 2, maxEOF: 0, maxBOF: 0 }],
+    'no-native-reassign': ON,
+    'no-new-require': ON,
+    'no-new-symbol': ON,
+    'no-obj-calls': ON,
+    'no-octal': ON,
+    'no-path-concat': ON,
+    'no-proto': ON,
+    'no-redeclare': [ON, { builtinGlobals: false }],
+    'no-restricted-globals': [ON, 'error', 'event'],
+    'no-restricted-modules': [ON, 'sys'],
+    'no-return-await': ON,
+    'no-script-url': ON,
+    'no-self-assign': ON,
+    'no-self-compare': ON,
+    'no-sequences': ON,
+    'no-setter-return': ON,
+    'no-shadow-restricted-names': ON,
+    'no-tabs': ON,
+    'no-template-curly-in-string': ON,
+    'no-this-before-super': ON,
+    'no-throw-literal': ON,
+    'no-trailing-spaces': ON,
+    'no-undef-init': ON,
+    'no-undef': [ON, { typeof: true }],
+    'no-unexpected-multiline': ON,
+    'no-unreachable': ON,
+    'no-unsafe-finally': ON,
+    'no-unsafe-negation': ON,
+    'no-unused-expressions': OFF,
+    'no-unused-labels': ON,
+    'no-unused-vars': [ON, { args: 'none', caughtErrors: 'all' }],
+    'no-use-before-define': [
+      ON,
+      {
+        classes: true,
+        functions: false,
+        variables: false,
+      },
+    ],
+    'no-useless-call': ON,
+    'no-useless-catch': ON,
+    'no-useless-concat': ON,
+    'no-useless-constructor': ON,
+    'no-useless-escape': ON,
+    'no-useless-return': ON,
+    'no-var': ON,
+    'no-void': ON,
+    'no-whitespace-before-property': ON,
+    'no-with': ON,
+    'object-curly-spacing': [ON, 'always'],
+    'object-shorthand': [ON, 'properties', { avoidQuotes: true }],
+    'one-var-declaration-per-line': ON,
+    'one-var': [ON, { initialized: 'never' }],
+    'operator-linebreak': [OFF, 'after'], // Off; conflicts with Prettier.
+    'prefer-const': [ON, { ignoreReadBeforeAssign: true }],
+    // 'quote-props': [ON, 'consistent'], // TODO
+    quotes: [ON, 'single', { avoidEscape: true }],
+    radix: ON,
+    'rest-spread-spacing': ON,
+    'semi-spacing': ON,
+    semi: [ON, 'always'],
+    'sort-keys': OFF, // TODO
+    'space-before-blocks': [ON, 'always'],
+    'space-before-function-paren': [
+      OFF, // Conflicts w/ Prettier.
+      {
+        anonymous: 'never',
+        named: 'never',
+        asyncArrow: 'always',
+      },
+    ],
+    'space-in-parens': [ON, 'never'],
+    'space-infix-ops': ON,
+    'space-unary-ops': ON,
+    'spaced-comment': [
+      ON,
+      'always',
+      {
+        block: { balanced: false },
+        exceptions: ['-', '**'],
+      },
+    ],
+    strict: [ON, 'global'],
+    'symbol-description': ON,
+    'template-curly-spacing': ON,
+    'unicode-bom': ON,
+    'use-isnan': ON,
+    'valid-typeof': ON,
+
+    // -------------------------------
+    // TypeScript ESLint Plugin Rules
+    // -------------------------------
+
+    '@typescript-eslint/array-type': [ON, {
```

Comment:
@ljharb, these would basically be the only rules that I use that need the TypeScript ESLint parser.

I'm not trying to be difficult, but it's possible that there is another way to do this that doesn't need to use this parser.

I'm hoping that these can all be replaced with some other plugin so we can use the parser you suggest.

Reply:
There is so little JavaScript in this repo right now, that basically the majority of it is going to be for linting the crawler, which I am in the process of developing. We don't really have to agree, but for my own purposes, these are my preferences. I will try to trim it down to bare necessities within the coming days. We may end not needing the majority of these anyway. 😂

Reply:
We definitely don't have to agree :-) i don't make the decisions on this repo either, to be clear.

If @codehag and others think it's fine to use jsdoc, and also think the tooling complexity to verify it is worth it, then your only real options are to do this, or to use `tsc` instead of eslint to typecheck the jsdoc separately (eslint-plugin-react [does this](https://github.com/yannickcr/eslint-plugin-react/blob/master/tsconfig.json)).

Reply:
As I've mentioned elsewhere, I don't think that the tooling complexity that this brings is worth it. If you want to use these tools during your work on the crawler, its fine -> you can create a dedicated eslint config for that folder that can be reviewed with those changes. For the website however, this is too much and we should use defaults. The website should have minimal JS code as it is a static content website.